src/main/java/org/mule/extension/runtime/ConfigurationInstanceRegistrationCallback.java

```diff
@@ -22,10 +22,11 @@
     /**
      * Registers the given {@code configurationInstance}
      *
-     * @param extension                 The {@link Extension} model for the {@code configurationInstance}
-     * @param configurationInstanceName a unique name for the {@code configurationInstance}
-     * @param configurationInstance     an instance that is a realisation of the {@link Configuration} model
-     * @param <C>                       the generic type of the {@code configurationInstance}
+     * @param extension                         The {@link Extension} model for the {@code configurationInstance}
+     * @param configurationInstanceProviderName the name of a registered {@link ConfigurationInstanceProvider} which owns the instance
+     * @param configurationInstance             an instance that is a realisation of the {@link Configuration} model
+     * @param <C>                               the generic type of the {@code configurationInstance}
+     * @return the name under which the {@code configurationInstance} was registered on the mule registry
```

Comment:
configurations have a lifecycle and they are registered. Since this change is 4.x only and we're assuming that ext-api will know the mule-api, it doesn't make sense to abstract the concept of the MuleRegistry since that should be the way of handling registration and lifecycle